app/services/request_password_reset.rb

```diff
@@ -16,15 +17,25 @@ def perform
   private
 
   def send_reset_password_instructions
+    irs_params = {
+      email: email,
+    }
+
     if Throttle.new(user: user, throttle_type: :reset_password_email).throttled_else_increment?
       analytics.throttler_rate_limit_triggered(throttle_type: :reset_password_email)
+      irs_params[:success] = false
+      irs_params[:failure_reason] = 'throttle rate limit triggered'
     else
       token = user.set_reset_password_token
       UserMailer.reset_password_instructions(user, email, token: token).deliver_now_or_later
 
       event = PushNotification::RecoveryActivatedEvent.new(user: user)
       PushNotification::HttpPush.deliver(event)
+
+      irs_params[:success] = true
```

Comment:
It's worth noting that the success is presumed here. The email might be delivered async.